### DISCUSSION TITLE
Metadata Node

### DISCUSSION THREAD

**Discussion by JustMaier** [WASasquatch/was-node-suite-comfyui#114] [Ideas]
Comfy's metadata is fantastic. Being able to describe a complex workflow as json is just so handy. However, because they can be so complex, they don't cater well to people working in simpler interfaces that only support basic workflows.

To try and cater to those users, as part of Civitai, we've added a [parser for workflows](https://github.com/civitai/civitai/blob/main/src/utils/metadata/comfy.metadata.ts) and tries to determine the basic params (prompt, negative, cfg, etc), it seems to be working for basic workflows, but there's lots of room for error.

It would be great if there was a Metadata Node that would embed basic generation params into a separate prop (perhaps `legacy_prompt` or something), that would allow users to pipe in the values they'd like to include.

Currently on Civit we expose the following params:
```
positive
negative
cfg
steps
seed
sampler
scheduler
denoise
width
height
model
```

**Comment by WASasquatch** [comment]
It seems the JSON may be scrubbed in the future of irrelevant data, though I suggested a "extra_info" object that could be used to store whatever from custom nodes. 

In light of that, I'm not sure how to proceed. Maybe an entirely different object, and like you said, called "legacy_prompt" which uses a more basic JSON object

---

I think a PR may also be necessary for the CLIPTextEncodes to append the raw prompt to a list for each time it encodes a prompt. Then these prompts can be retrieved as a simple list that contains the prompts encoded in order (when reversed).
